azext_edge/edge/providers/orchestration/work.py

```diff
@@ -368,6 +378,19 @@ def _do_work(self):
                     verify_write_permission_against_rg(
                         subscription_id=self.subscription_id, resource_group_name=self._targets.resource_group_name
                     )
+                if self._check_cluster:
+                    load_config_context(context_name=self._context_name)
+                    validate_cluster_prechecks(
```

Comment:
What if we did a dynamic kwargs strategy to pass into validate_cluster_prechecks(), removing inline if else conditionals, whose benefit grows as more parameters are used at the same time.

something like
```python
cluster_precheck_kwargs = {}
if self._targets.enable_fault_tolerance:
  cluster_precheck_kwargs["acs_config"] =  get_merged_acs_config(...)
validate_cluster_prechecks(**cluster_precheck_kwargs)
```

Reply:
to kwarg or not to kwarg, that is the question :)

i like this strategy, will push an update soon with `check_kwargs` being built in a helper method to organize the logic